app/src/endpoints.c

```diff
@@ -282,6 +320,17 @@ static bool is_ble_ready(void) {
 }
 
 static enum zmk_transport get_selected_transport(void) {
+#if IS_ENABLED(CONFIG_ZMK_ENDPOINT_DISABLE_FALLBACK)
+    switch (preferred_transport) {
+    case ZMK_TRANSPORT_BLE:
+        return is_ble_ready() ? ZMK_TRANSPORT_BLE : ZMK_TRANSPORT_NONE;
+    case ZMK_TRANSPORT_USB:
+        return is_usb_ready() ? ZMK_TRANSPORT_USB : ZMK_TRANSPORT_NONE;
+
+    default:
```

Comment:
I'm not a fan of default cases with no logic. If you omit the default case, then the compiler can verify that you have a case for every enum value. Then, if you add a new enum value later but forget to update this code, it will give a diagnostic. 

I suggest having a case for ZMK_TRANSPORT_NONE which just returns ZMK_TRANSPORT_NONE, and then treating it as an error if none of the cases are hit. Maybe even split this into two completely separate implementations of the function like this?

```c
#if IS_ENABLED(CONFIG_ZMK_ENDPOINT_DISABLE_FALLBACK)

static enum zmk_transport get_selected_transport(void) {
    switch (preferred_transport) {
        case ZMK_TRANSPORT_BLE: 
            return ...
        case ZMK_TRANSPORT_USB: 
            return ...
        case ZMK_TRANSPORT_NONE: 
            return ZMK_TRANSPORT_NONE;
    }

    LOG_ERR("Unknown transport %d", preferred_transport);
    return ZMK_TRANSPORT_NONE;
}

#else

// original implementation

#endif
```

Reply:
Both of these suggestions have now been implemented